pkg/virt-controller/watch/clone/clone_test.go

```diff
@@ -459,6 +467,60 @@ var _ = Describe("Clone", func() {
 				expectEvent(TargetVMCreated)
 			})
 
+			When("the clone process is finished and involves one or more PVCs", func() {
+				var (
+					pvc      *k8sv1.PersistentVolumeClaim
+					snapshot *snapshotv1alpha1.VirtualMachineSnapshot
+					restore  *snapshotv1alpha1.VirtualMachineRestore
+				)
+
+				BeforeEach(func() {
+					snapshot = createVirtualMachineSnapshot(sourceVM)
+					snapshot.Status.ReadyToUse = pointer.Bool(true)
+
+					pvc = createPVC(sourceVM.Namespace, k8sv1.ClaimPending)
+					restore = createVirtualMachineRestore(sourceVM, snapshot.Name)
+					restore.Status.Complete = pointer.Bool(true)
+					restore.Status.Restores = []snapshotv1alpha1.VolumeRestore{
+						{PersistentVolumeClaimName: pvc.Name},
+					}
+
+					vmClone.Status.SnapshotName = pointer.String(snapshot.Name)
+					vmClone.Status.RestoreName = pointer.String(restore.Name)
+					vmClone.Status.Phase = clonev1alpha1.CreatingTargetVM
+
+					targetVM := sourceVM.DeepCopy()
+					targetVM.Name = vmClone.Spec.Target.Name
+
+					addVM(sourceVM)
+					addVM(targetVM)
+					addClone(vmClone)
+					addSnapshot(snapshot)
+					addRestore(restore)
+					addPVC(pvc)
+
+					expectCloneUpdate(clonev1alpha1.Succeeded)
+					controller.Execute()
+					expectEvent(TargetVMCreated)
+				})
+
+				It("if not all the PVCs are bound, nothing should happen", func() {
```

Comment:
Maybe we can issue a GET/LIST just to make sure "nothing happened"

Reply:
Thank you @akalenyu! I agree w/ you and I was creating this(https://github.com/kubevirt/kubevirt/commit/6f892bea3c31f02f0fa04443275e51db07660795) followup commit containing a refactoring for the whole vm clone unit tests.
I did not include it in this PR because it will result in too many things in one PR IMO.
If you think that it's worth it to add it in this PR, I will do it. WDYT? Thanks

Reply:
Very interesting; usually, I don't see this type of testing a lot in this repo, and I am not sure what's the guideline.
(Usually kubevirt unit tests use reactors)
Definitely not in this PR though, for this one, I was thinking about a negating expectDelete or something along those lines

Reply:
Well, expecting something not to happen here is very hard. :)
How about `expectNoEvents`?

Reply:
You can have snapshot delete reactors that just Fail() the test

Reply:
We already have a base common
```
client.Fake.PrependReactor("*", "*", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
	Expect(action).To(BeNil())
	return true, nil, nil
})
```
that will fail the test if any unexpected api call is fired :)

Reply:
Great! so that is good enough

Reply:
Thank you!
Will mark the conversation as resolved :)